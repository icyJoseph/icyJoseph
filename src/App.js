import Root from "./routes";
import BSOD from "./containers/BSOD";

// check if the userAgent is IE
export function browserIE() {
  const userAgent = window.navigator.userAgent;
  return /MSIE|Trident/.test(userAgent);
}

// run the app
export function run() {
  const userHasIE = browserIE();
  if (userHasIE) {
    return BSOD;
  }

  return Root;
}

export default run;
