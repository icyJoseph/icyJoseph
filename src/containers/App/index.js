import React, { Component } from "react";
import PropTypes from "prop-types";

export class App extends Component {
  render() {
    const { match } = this.props;
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to {match.path}</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default App;

App.propTypes = {
  location: PropTypes.object,
  match: PropTypes.object,
  history: PropTypes.object
};
