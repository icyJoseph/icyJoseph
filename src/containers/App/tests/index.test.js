import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import App from "../";

describe("App", () => {
  const props = {
    match: { path: "/" }
  };
  const wrapper = shallow(<App {...props} />);
  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
