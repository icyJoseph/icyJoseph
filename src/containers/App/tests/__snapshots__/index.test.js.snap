// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    match={
        Object {
            "path": "/",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Welcome to 
                    /
          </h1>
</header>,
        <p
          className="App-intro"
>
          To get started, edit 
          <code>
                    src/App.js
          </code>
           and save to reload.
</p>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Welcome to 
            /
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome to ",
              "/",
            ],
            "className": "App-title",
          },
          "ref": null,
          "rendered": Array [
            "Welcome to ",
            "/",
          ],
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.js
</code>,
            " and save to reload.",
          ],
          "className": "App-intro",
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.js",
            },
            "ref": null,
            "rendered": "src/App.js",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Welcome to 
                        /
            </h1>
</header>,
          <p
            className="App-intro"
>
            To get started, edit 
            <code>
                        src/App.js
            </code>
             and save to reload.
</p>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Welcome to 
              /
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Welcome to ",
                "/",
              ],
              "className": "App-title",
            },
            "ref": null,
            "rendered": Array [
              "Welcome to ",
              "/",
            ],
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.js
</code>,
              " and save to reload.",
            ],
            "className": "App-intro",
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.js",
              },
              "ref": null,
              "rendered": "src/App.js",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
